tests/rptest/tests/prefix_truncate_recovery_test.py

```diff
@@ -78,7 +77,6 @@ def get_segments_deleted(self, nodes):
         """
         metric = self.redpanda.metrics_sample("log_segments_removed", nodes)
         metric = metric.label_filter(dict(namespace="kafka", topic=self.topic))
-        assert len(metric.samples) == len(nodes)
```

Comment:
this one suffers from the same edge case:
in produce_util_reclaim we would stop if the resulting list is empty, 
and in run_recovery we would terminate with this

```
>>> max([])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: max() arg is an empty sequence

```

Reply:
I have removed this change from the PR for now.